flutter-studio/src/io/flutter/profiler/FilterLibraryDialog.java

```diff
@@ -0,0 +1,199 @@
+      @Override
+      public void mouseClicked(MouseEvent e) {
+        popupDialog.checkBoxList.setCheckStateAllItems(true);
+        popupDialog.checkBoxList.updateUI();
+      }
+    });
+
+    CommonButton uncheck = new CommonButton("Uncheck All");
+    uncheck.setForeground(Color.BLACK);
+    uncheck.setBorder(buttonBorder);
+    uncheck.addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent e) {
+        popupDialog.checkBoxList.setCheckStateAllItems(false);
+        popupDialog.checkBoxList.updateUI();
+      }
+    });
+
+    // Check the libraries that are being filtered.
+    popupDialog.checkBoxList.setCheckStateAllItems(previousSelectedLibraries);
+
+    JPanel buttonPanel = new JPanel();
+    buttonPanel.setLayout(new FlowLayout(FlowLayout.LEFT, 50, 0));
+    buttonPanel.add(check);
+    buttonPanel.add(uncheck);
+    panel.add(buttonPanel);
+
+    return panel;
+  }
+}
+
+/**
+ * Handle filtering of which Dart classes based on libraries
+ */
+class PopupLibraryFilter extends JPanel {
+  DefaultListModel<JCheckBox> model;
+  JCheckBoxList checkBoxList;
+
+  PopupLibraryFilter(String[] libraries) {
+    model = new DefaultListModel<JCheckBox>();
+
+    for (int index = 0; index < libraries.length; index++) {
+      model.addElement(new JCheckBox(libraries[index]));
+    }
+
+    checkBoxList = new JCheckBoxList(model);
+
+    JScrollPane scrollPane = new JBScrollPane(checkBoxList);
+    this.add(scrollPane);
+
+    this.setVisible(true);
+  }
+
+  List<String> librariesToFilter() {
+    List<String> checkedLibraries = new ArrayList<>();
+
+    for (int index = 0; index < model.getSize(); index++) {
+      JCheckBox cb = model.get(index);
+      if (cb.isSelected()) {
+        checkedLibraries.add(cb.getText());
+      }
+    }
+
+    return checkedLibraries;
+  }
+
+  protected class JCheckBoxList extends JList<JCheckBox> {
+    protected final Border noFocusBorder = new EmptyBorder(1, 1, 1, 1);
+
+    public JCheckBoxList() {
+      setCellRenderer(new CellRenderer());
+      addMouseListener(new MouseAdapter() {
+        @Override
+        public void mousePressed(MouseEvent e) {
+          int index = locationToIndex(e.getPoint());
+          if (index != -1) {
+            JCheckBox checkbox = (JCheckBox)getModel().getElementAt(index);
+            checkbox.setSelected(!checkbox.isSelected());
+            repaint();
+          }
+        }
+      });
+      setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+    }
+
+    public JCheckBoxList(ListModel<JCheckBox> model) {
+      this();
+      setModel(model);
+    }
+
+    public void setCheckStateAllItems(boolean isChecked) {
+      int modelSize = this.getModel().getSize();
+      for (int index = 0; index < modelSize; index++) {
+        this.getModel().getElementAt(index).setSelected(isChecked);
+      }
+    }
+
+    public void setCheckStateAllItems(List<String> librariesSelected) {
```

Comment:
nit: is there a reason librariesSelected isn't
Set<String> to make this method more efficient?

Reply:
None, I suspect I was using List<String> as I was working on the code, moved to the set for filtering but never changed the initial list used by the dialog to set.  Done.